Merge pointer tests and protected

tannerblair opened the pull request:
- [x] This contribution adheres to [contributing.md](docs/contributing.md).

### What does this Pull Request accomplish?

Adds unit tests for the Memory Manager API.

Introduces a Protected Pointer subclass to remove lock from main class.

### Why should this Pull Request be merged?

Because unit tests are a good idea

### What testing has been done?

Unit Tests have been created for all of the library's public functions.

Commit:
Added lifecycle tests and Read tests for Pointer.lvclass

Commit:
fixed directories for existing tests, created mem copy test

Commit:
separated compiled code from source

Commit:
merging in Protected Pointer class

Commit:
added clear tests

Commit:
finished data manipulation tests

Commit:
Merge branch 'v1.0.0' into merge-pointer-tests-and-protected

csjall commented:
- [ ] Collapse the folder structure by removing LabVIEW memory manager folder. If you add additional projects in the future, then create the disambiguating sub folders. 
- [ ] Move license file to root level. This allows GitHub to preview the license.
- [ ] AppSW uses a stable mainline branching strategy where all changes are committed to master and versioned releases are branched from master. Consider using this strategy moving forward.
https://www.bitsnbites.eu/a-stable-mainline-branching-model-for-git/
- [ ] Consider using [LabVIEWCLI](https://zone.ni.com/reference/en-XX/help/371361R-01/lvhowto/cli_running_operations/) for your build process. It's relatively simple to create custom extension. The big bonus is that you can use command line with build pipeline. One of the first things we did is convert all our building and testing scripts to LabVIEWCLI for the custom devices. See [RunVITester](https://github.com/ni/niveristand-custom-device-testing-tools/tree/master/RunVITester) for an example of writing a custom extension.
- [x] Use consistent naming for your tests. I recommend "test <Verb> <Noun> naming scheme (e.g. "test Read Boolean".
![image](https://user-images.githubusercontent.com/4081815/63785357-da61f780-c8b5-11e9-8681-00e772ab02c2.png)
- [x] Remove example test case, temp_VI_UnderTest.vi, from test classes. It was code generated as an example. You can safely remove it.
- [x] Some tests do not clean up pointers. Test should be cleaning up after themselves and not leave stale resources test to test.
![image](https://user-images.githubusercontent.com/4081815/63786077-08940700-c8b7-11e9-8b38-b7fb6724f917.png)
- [x] Clean up overlapping wires.
![image](https://user-images.githubusercontent.com/4081815/63786890-83a9ed00-c8b8-11e9-8089-c1860d0f50d8.png)
- [x] The Write/ClearMem bug is rather disconcerting. You should figure out what is going with that OR simply zero the memory.

Commit:
made "dest" and "size" parameters on Move Block non-constant.

Commit:
updating methods affected by changing MoveBlock

Commit:
updated write methods to use the right method template for moveblock

Commit:
added tests to check that data on the wire isn't changed when a pointer is written

Commit:
cleaned up test names and removed stub test vis

tannerblair commented:
@csjall, with regards to the following items:

> Collapse the folder structure by removing LabVIEW memory manager folder. If you add additional projects in the future, then create the disambiguating sub folders.

> Move license file to root level. This allows GitHub to preview the license.

> AppSW uses a stable mainline branching strategy where all changes are committed to master and versioned releases are branched from master. Consider using this strategy moving forward.
https://www.bitsnbites.eu/a-stable-mainline-branching-model-for-git/

> Consider using LabVIEWCLI for your build process. It's relatively simple to create custom extension. The big bonus is that you can use command line with build pipeline. One of the first things we did is convert all our building and testing scripts to LabVIEWCLI for the custom devices. See RunVITester for an example of writing a custom extension.

I agree with all of these, and I've created items in the backlog for all of them. I would prefer to do them in a separate PR after @jag2k2 and I have merged our code back together. Let me know if you have any concerns, but otherwise I believe I've addressed all of the issues above.

Commit:
cleaned up wires and ensured that all pointers are destroyed

Commit:
removed unused boolean in class private data for tests

Commit:
fixed mistake with error wire in read tests

Commit:
removed invalid destroy from test

csjall commented:
The folder layout changes shouldn't hold up this review. Moving the cleanup items to the backlog seems reasonable.

I am impressed with your test coverage. Once you get an automated build and test run going you will have much greater confidence when making changes to this code. You can always add additional tests as needed, but this is a solid foundation to start from.

Commit:
Merge branch 'v1.0.0' into merge-pointer-tests-and-protected

# Conflicts:
#	LabVIEW Memory Manager API/Tests/Deterministic Error Tests/Deterministic Error Tests.lvlib
